### ISSUE TITLE
False Positive ERROR thrown when this.fn is returned in a closure bound to `this`

### ISSUE THREAD

**Issue by philn5d** [palantir/tslint#3658]
### Bug Report

- __TSLint version__: tslint@5.8.0
- __TypeScript version__: typescript@2.6.2
- __Running TSLint via__: VSCode npm

#### TypeScript code being linted

```ts
class ClosureIssue {
    getCallback (in1) {
        return function (a) {
            this.doCallback1(a)
        }.bind(this)
    }
    private doCallback1(a) {
        console.log('log me up Scotty')
    }
}

```

with `tslint.json` configuration:

```json
{
    "rules": {
      "class-name": true,
      "comment-format": [
        true,
        "check-space"
      ],
      "curly": false,
      "eofline": true,
      "forin": true,
      "indent": [
        true,
        "spaces"
      ],
      "label-position": true,
      "max-line-length": [
        true,
        140
      ],
      "member-access": false,
      "member-ordering": [
        true,
        "static-before-instance",
        "variables-before-functions"
      ],
      "no-arg": true,
      "no-bitwise": true,
      "no-console": [
        true,
        "debug",
        "info",
        "time",
        "timeEnd",
        "trace"
      ],
      "no-construct": true,
      "no-debugger": true,
      "no-duplicate-variable": true,
      "no-empty": false,
      "no-eval": true,
      "no-inferrable-types": true,
      "no-shadowed-variable": true,
      "no-string-literal": false,
      "no-switch-case-fall-through": true,
      "no-trailing-whitespace": true,
      "no-unused-expression": true,
      "no-unused-variable": true,
      "no-use-before-declare": true,
      "no-var-keyword": true,
      "object-literal-sort-keys": false,
      "one-line": [
        true,
        "check-open-brace",
        "check-catch",
        "check-else",
        "check-whitespace"
      ],
      "quotemark": [
        true,
        "single"
      ],
      "radix": true,
      "semicolon": [
        "always"
      ],
      "triple-equals": [
        true,
        "allow-null-check"
      ],
      "typedef-whitespace": [
        true,
        {
          "call-signature": "nospace",
          "index-signature": "nospace",
          "parameter": "nospace",
          "property-declaration": "nospace",
          "variable-declaration": "nospace"
        }
      ],
      "variable-name": false,
      "whitespace": [
        true,
        "check-branch",
        "check-decl",
        "check-operator",
        "check-separator",
        "check-type"
      ]
    }
  }
```

#### Actual behavior
sample-ts-lint-issue.ts[7, 13]: 'doCallback1' is declared but its value is never read.
#### Expected behavior
no error reported as "is declared but its value is never read." since the method is read from within the closure returned from `getCallback`.

**Comment by JoshuaKGoldberg** [comment]
Ooh, this is a fun one. We _could_ add a setting to the rule to treat `.bind(this)` as implying the `function` has a `this` scope set to the binder... but @philn5d how about using an arrow lambda instead?

Technically, the change is doable, but seems like it wouldn't be worth the maintenance cost. Thoughts?

**Comment by philn5d** [comment]
I'm simply the messenger reporting a bug. 

Consider: what's the impact of a developer deleting this type of "unused" function? I'm not a TS expert, but it seems like it would result in a run-time error. If it's a compile-time error, OK the transpiler will catch it - not that much of a problem. Still, it is incorrect to say it's unused.

**Comment by JoshuaKGoldberg** [comment]
`no-unused-variable` is now deprecated in TSLint core; if and when it gets re-added, it's likely to use the native TypeScript APIs. #4100 

Thanks for posting this! Certainly an interesting edge case to think about 😄 and perhaps an issue with TypeScript itself.
